app/Controllers/feedController.php

```diff
@@ -647,6 +647,150 @@ public function deleteAction() {
 		}
 	}
 
+	/**
+	 * This action force clears the cache of a feed.
+	 *
+	 * Parameters are:
+	 *   - id (mandatory - no default): Feed ID
+	 *
+	 */
+	public function clearcacheAction() {
+		//Get Feed.
+		$id = Minz_Request::param('id');
+
+		$feedDAO = FreshRSS_Factory::createFeedDao();
+		$feed = $feedDAO->searchById($id);
+
+		if (!$feed) {
+			Minz_Request::bad(_t('feedback.sub.feed.not_found'), array());
+			return;
+		}
+
+		$feed->clearCache();
+
+		Minz_Request::good(_t('feedback.sub.feed.cache_cleared', $feed->name()), array(
+			'params' => array('get' => 'f_' . $feed->id())
+		));
+	}
+
+	/**
+	 * This action forces reloading the articles of a feed.
+	 *
+	 * Parameters are:
+	 *   - id (mandatory - no default): Feed ID
+	 *
+	 */
+	public function reloadAction() {
+		@set_time_limit(300);
+
+		//Get Feed ID.
+		$feed_id = Minz_Request::param('id');
+
+		$feedDAO = FreshRSS_Factory::createFeedDao();
+		$entryDAO = FreshRSS_Factory::createEntryDao();
+
+		$feed = $feedDAO->searchById($feed_id);
+
+		if (!$feed) {
+			Minz_Request::bad(_t('feedback.sub.feed.not_found'), array());
+			return;
+		}
+
+		//Re-fetch articles as if the feed was new.
+		self::actualizeFeed($feed_id, null, false, null, true);
+
+		//Extract all feed entries from database, load complete content and store them back in database.
+		$entries = $entryDAO->listWhere('f', $feed_id, FreshRSS_Entry::STATE_ALL, 'DESC', 0);
+
+		$entryDAO->beginTransaction();
+
+		foreach ($entries as $entry) {
+			$entry->loadCompleteContent(true);
+			$entryDAO->updateEntry($entry->toArray());
+		}
+
+		$entryDAO->commit();
+
+		//Give feedback to user.
+		Minz_Request::good(_t('feedback.sub.feed.reloaded', $feed->name()), array(
+			'params' => array('get' => 'f_' . $feed->id())
+		));
+	}
+
+	/**
+	 * This action creates a preview of a content-selector.
+	 *
+	 * Parameters are:
+	 *   - id (mandatory - no default): Feed ID
+	 *   - selector (mandatory - no default): Selector to preview
+	 *
+	 */
+	public function contentSelectorPreviewAction() {
+
+		//Configure.
+		$this->view->fatalError = '';
+		$this->view->selectorSuccess = false;
+		$this->view->htmlContent = '';
+
+		$this->view->_layout(false);
+
+		//XXX 'script-src' allow possible script inlined in the previewed page to be executed.
+		//If we want to allow only our script, perhaps we should create a file, and reference it, instead of having it in-line.
+		//This view will be in an iframe, without any context, is it that dangerous? What a script can do?
```

Comment:
Actually the risk is pretty low since we sanitize the content. I would avoid this anyway since it doesn't seem required (if my previous suggestion works :)). Here I would reuse the policies of the index page to be consistent:

```php
		$this->_csp([
			'default-src' => "'self'",
			'frame-src' => '*',
			'img-src' => '* data:',
			'media-src' => '*',
		]);
```